llvm/lib/CodeGen/WindowScheduler.cpp

```diff
@@ -485,15 +485,17 @@ int WindowScheduler::calculateMaxCycle(ScheduleDAGInstrs &DAG,
 // ========================================
 int WindowScheduler::calculateStallCycle(unsigned Offset, int MaxCycle) {
   int MaxStallCycle = 0;
+  int CurrentII = MaxCycle + 1;
```

Comment:
Please provide a schematic diagram for the calculation of stall cycles.

Reply:
To make the changes in this PR much clear, here is an example of `StallCycle` calculation. 
> Without adding `1` to `MaxCycle`, `StallCycle` would be `4` instead of `3`, which is not correct. 
![image](https://github.com/user-attachments/assets/fdbb093d-9ae9-4fe5-88e1-20ba8a80fd01)

Reply:
LGTM